src/script/engine.zig

```diff
@@ -93,56 +82,56 @@ pub const Engine = struct {
         }
     }
 
-    /// Execute a single opcode
-    ///
-    /// # Arguments
-    /// - `opcode`: The opcode to execute
-    ///
-    /// # Returns
-    /// - `EngineError`: If an error occurs during execution
     fn executeOpcode(self: *Engine, opcode: u8) !void {
```

Comment:
Can you make it so that this method takes as argument `opcode: Opcode` so that it can only be called with valid value.

Then you add a conversion u8 -> !Opcode just before calling it, in `execute`.